crates/bevy_math/src/cubic_splines.rs

```diff
@@ -23,25 +25,42 @@ pub trait Point:
     + Copy
 {
 }
-impl Point for Vec3 {}
-impl Point for Vec3A {}
-impl Point for Vec2 {}
-impl Point for f32 {}
+
+impl<T> Point for T where
+    T: Mul<f32, Output = Self>
+        + Div<f32, Output = Self>
+        + Add<Self, Output = Self>
+        + Sub<Self, Output = Self>
+        + Add<f32, Output = Self>
+        + Sum
+        + Default
+        + Debug
+        + Clone
+        + PartialEq
+        + Copy
+{
+}
```

Comment:
I'd like to argue for reverting this change, though I see its value.

There are a few good reasons not to blanket impl, and instead use concrete impls:

1. It's easier to figure out what the intended use is. This isn't a very strong argument, considering we have examples, and can provide docs, but having a finite set of types you can use makes it simpler to understand what you are supposed to plug into those generics, and the compiler will suggest these for you, something like `these types implement Point: Vec3, ...`.
2. Blanket impls will negatively impact compile times. This is a pretty broad blanket impl, many math types across many crates could potentially be used here, and all of the generic functions will need to be monomorphised for every single matching type. It's hard to quantify, but blanket impls can make the final compilation step take a long time as they accumulate in your dependency tree.
3. `bevy_math` is already opinionated about its math types - we use `glam`, I think it's reasonable then to expect that our spline types also only support a set of glam types.
4. Users will still have the escape hatch of newtyping and implementing the `Point` trait on their newtype.

If anything, I think we just need to expand the set of glam types that we implement this for, like the integer types, if applicable.

Reply:
I agree: this should be reverted.

Reply:
I do not agree with the approach of not using a blanket implementation, as it's simply a marker trait with no methods. It's simply a collection of other math traits required.
Using a newtype is, of course, a valid approach but I personally don't like it. I see and understand the arguments for it, which is why I reverted the change.
But hey, at some point in collaboration compromises will have to be made.

Reply:
The compile-time argument is compelling, but without the blanket impl we are playing a dangerous game with rust's orphan rule. It's all well and good to implement `Point` on your own types, but what happens when a user needs to implement it on numeric type from an external crate?

Reply:
@NthTensor The 4th point addresses exactly that. Users can make a newtype wrapping the target type and implement `Point` on that. But I just realized that it would also require implementing all the dependencies of the main trait, which is a lot of boilerplate in my opinion.

Reply:
I am currently more inclined to un-revert teh blanket impl and add some documentation to list common types, probably at module level.

Reply:
I can live with that. The orphan rule is a compelling argument.

Reply:
I would like to know what @aevyrie thinks before I un-revert this

Reply:
I'm not the arbiter of what gets merged here, just a person with an opinion. 🙂 

I have to push back on "orphan rule is hard" - why would we encourage users to use non-glam types here, when `bevy_math` forces you to use `glam` *everywhere* else? Following that path leads you to the hyper-generic errors you see in nalgebra-like crates. The only reason the `Point` trait exists is to reduce code duplication across the 1d/2d/3d/SIMD `glam` types. If a user has a non-glam type they want to use here, they are *already* forced to solve the `Into<glam>` conversions everywhere else in bevy, so this isn't introducing some new burden for them.

If anything, I regret making the trait public at all, it's ultimately an implementation detail, and isn't something that needs to be generic across any types other than the glam types we want to support.

My argument is essentially that the orphan rule difficulty is a non-sequitor, we don't *want* to support users plugging arbitrary types into `bevy_math`, that is over-generalization.

Reply:
>  it's simply a marker trait with no methods

Yes, but every generic `<P: Point>` function and type in this module will be monomorphised. I realize this could be small on its own, but bad compile times are death by a thousand cuts, and I don't see any utility in allowing non `glam` math types in this one place.